src/ParseServer.js

```diff
@@ -323,6 +329,26 @@ class ParseServer {
     return parseServer.start(options, callback);
   }
 
+  /**
+   * @static
+   * Create an async parse server
+   * @param {ParseServerOptions} options the parse server initialization options 
+   */
+  static createAsync(options) {
+    return new Promise((resolve, reject) => {
+      const parseServer = new ParseServer({
+        ...options,
+        serverStartComplete(error) {
+          if (error) {
+            reject(error);
```

Comment:
I know that the original use case of `serverStartComplete` will have less sense here.
`serverStartComplete` is still a valid option for an `startAsync`  parse-server.

You need to try to call the original "serverStartComplete()" before on reject or resolve, if `options. serverStartComplete` exists.

Then developers will not have weird behavior, (currently you overwrite the serverStartComplete) if they use startAsync.